api/src/paths/alert/index.ts

```diff
@@ -0,0 +1,244 @@
+import { Request, RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { SYSTEM_ROLE } from '../../constants/roles';
+import { getDBConnection } from '../../database/db';
+import { IAlertFilterObject } from '../../models/alert-view';
+import { systemAlertCreateSchema, systemAlertSchema } from '../../openapi/schemas/alert';
+import { authorizeRequestHandler } from '../../request-handlers/security/authorization';
+import { AlertService } from '../../services/alert-service';
+import { getLogger } from '../../utils/logger';
+
+const defaultLog = getLogger('paths/alert/index');
+
+export const GET: Operation = [
+  authorizeRequestHandler(() => {
+    return {
+      and: [
+        {
+          discriminator: 'SystemUser'
+        }
+      ]
+    };
+  }),
+  getAlerts()
+];
+
+GET.apiDoc = {
+  description: 'Gets a list of system alerts.',
+  tags: ['alerts'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      in: 'query',
+      name: 'types',
+      required: false,
+      schema: {
+        type: 'array',
+        items: {
+          type: 'string',
+          nullable: true
```

Comment:
You probably dont want this to be nullable. Since this is inside the `items` section, this means that the array is allowed to have null values.
```
types=['type1', null, 'type2']
```
I would just remove the nullable entirely, and leave the query param as not required. That way the caller can omit `types`, or else include it with non-null values.

Same with the query params below.

Reply:
Fixed